File: Python/Linked List/LinkedList.py
class Node:
    # Constructors
    def __init__(self):
        self.data = 0
        self.next = None

    def __init__(self, data):
        self.data = data
        self.next = None
    
    def __init__(self, data, next):
        self.data = data
        self.next = next

class LinkedList:
    # Contstructors
    def __init__(self):
        self.head = None
        self.size = 0

    def __init__(self, head):
        self.head = head
        self.size = 0

    def __init__(self, list):
        self.head = Node(list[0])
        self.size = 1

        current = self.head
        for i in range(1, len(list)):
            current.next = Node(list[i])
            current = current.getNext()
            self.size += 1

    # Methods
    def push_front(self, data):
        self.head = Node(data, self.head)

    def push_back(self, data):
        current = self.head

        while current.next() != None:
            current = current.next
        
        current.next = Node(data)

    def insert(self, data, idx):
        if idx <= 0:
            self.push_front(data)
        elif idx >= self.size:
            self.push_back(data)
        else:
            current = self.head
            for i in range(idx):
                current = current.next

            current.next = Node(data, current.next)

    def remove(self, data):
        current = self.head
        while(current.next != None):
            if current.next.data == data:
                current.next = current.next.next
                return
            current = current.next

    def contains(self, data):
        current = self.head
        while current.next != None:
            if current.data == data:
                return True
            current = current.next
        return False
    
    def print(self):
        current = self.head
        while current != None:
            print(current.data + ' ')
            current = current.next

File: Java/linked_list/LinkedList.java
package linked_list;

public class LinkedList {
    private Node head;
    private int size;

    public LinkedList() {
        this.head = null;
        this.size = 0;
    }

    public LinkedList(int data) {
        this.head = new Node(data);
        this.size = 1;
    }

    public LinkedList (int[] list){
        this.head = new Node(list[0]);
        this.size = 1;
        for (int i = 1; i < list.length; i++) {
            this.append(list[i]);
            this.size++;
        }
    }

    public Node getHead() {
        return this.head;
    }

    public int getSize() {
        return this.size;
    }

    public void prepend(int data) {
        Node newNode = new Node(data);
        newNode.setNext(this.head);
        this.head = newNode;
        this.size++;
    }

    public void append(int data){
        Node newNode = new Node(data);
        Node current = this.head;
        while (current.getNext() != null) {
            current = current.getNext();
        }
        current.setNext(newNode);
        this.size++;
    }

    public void insert(int data, int idx){
        if (idx < 1) {
            this.prepend(data);
        }
        else if (idx >= this.size){
            this.append(data);
        }
        else {
            Node newNode = new Node(data);
            Node current = this.head;
            for (int i = 0; i < idx - 1; i++) {
                current = current.getNext();
            }
            newNode.setNext(current.getNext());
            current.setNext(newNode);
            this.size++;
        }
    }

    public void remove(int data){
        Node current = this.head;
        if (current.getData() == data) {
            this.head = current.getNext();
            this.size--;
        }
        else {
            while (current.getNext() != null) {
                if (current.getNext().getData() == data) {
                    current.setNext(current.getNext().getNext());
                    this.size--;
                    break;
                }
                current = current.getNext();
            }
        }
    }

    public boolean contains(int data){
        Node current = this.head;
        while (current != null) {
            if (current.getData() == data) {
                return true;
            }
            current = current.getNext();
        }
        return false;
    }

    public void printList(){
        Node current = this.head;
        while (current != null) {
            System.out.print(current.getData() + " ");
            current = current.getNext();
        }
        System.out.println();
    }
}
